### ISSUE TITLE
DRM

### ISSUE THREAD

**Issue by lawrencenull** [minbrowser/min#641]
I have been looking for a reasonabley simple way to add widevine to electron an came across https://github.com/castlabs/electron-releases from a Google partner. Has anyone tried to use this yet especially in the context of min?

I personally think Min only lacks a few things that could make it a daily driver, support for drm content and webrtc.

**Comment by PalmerAL** [comment]
I have no idea if this works or not, but we're not going to include it in the prebuilt releases regardless. According to that page, in order to include widevine you have to get a special license agreement from Google and have your package codesigned by them, which isn't something I really want to do (and it probably isn't feasible anyway). The fact that this is an Electron fork is also a problem, because it's probably not going to be updated very regularly.

The [Electon docs suggest](https://electronjs.org/docs/tutorial/testing-widevine-cdm) you can just use the Widevine binary included with Chrome. I'm not really sure if this is violating a license agreement or not, so I don't plan on doing it, but that might be something you could investigate.

WebRTC should be supported, the issue is that we don't support permission requests for camera/microphone/etc - that's issue #332.

**Comment by lawrencenull** [comment]
Thanks for the response. I can understand from the licensing point of view, I'll see if I can make a hack(y) try at it.

**Comment by dbfreem** [comment]
@palmeral it sounds like this should be closed or at a minimum renamed to reflect the webRTC part only. Thoughts?

**Comment by lawrencenull** [comment]
Agree, if DRM isn't in the future webrtc would still be a nice addition.

**Comment by PalmerAL** [comment]
@dbfreem The second issue isn't really webRTC (that's already supported), but that we don't support permission requests for camera and microphone, which there's already an issue for, so I think this issue is just the DRM part.

If we could figure out the licensing issues, I think it would be nice to offer DRM as an optional download (like Firefox does), so maybe this issue should be left open? But given articles like this: https://blog.samuelmaddock.com/posts/google-widevine-blocked-my-browser/, I think that's fairly unlikely.

**Comment by lpossamai** [comment]
+1 - 4 years later and still an open issue - it would be nice to have this taken care of

**Comment by aiyagari** [comment]
> I have no idea if this works or not, but we're not going to include it in the prebuilt releases regardless. According to that page, in order to include widevine you have to get a special license agreement from Google and have your package codesigned by them, which isn't something I really want to do (and it probably isn't feasible anyway). The fact that this is an Electron fork is also a problem, because it's probably not going to be updated very regularly.

It seems no longer necessary to have a specific license agreement with Google, since there is now a service for clients to generate keys:

https://github.com/castlabs/electron-releases/wiki/EVS

**Comment by sontopo** [comment]
> > I have no idea if this works or not, but we're not going to include it in the prebuilt releases regardless. According to that page, in order to include widevine you have to get a special license agreement from Google and have your package codesigned by them, which isn't something I really want to do (and it probably isn't feasible anyway). The fact that this is an Electron fork is also a problem, because it's probably not going to be updated very regularly.
> 
> It seems no longer necessary to have a specific license agreement with Google, since there is now a service for clients to generate keys:
> 
> https://github.com/castlabs/electron-releases/wiki/EVS

So no news about this? If you dont need an agreement with google anymore, maybe they should reconsider adding it.

**Comment by aiyagari** [comment]
In fact the castlabs project has become even more simplified, supplying sample integration code.
https://github.com/castlabs/electron-releases#how-does-it-work
